src/ring_theory/power_basis.lean

```diff
@@ -121,6 +121,25 @@ lemma exists_eq_aeval [nontrivial S] (pb : power_basis R S) (y : S) :
   ∃ f : polynomial R, f.nat_degree < pb.dim ∧ y = aeval pb.gen f :=
 (mem_span_pow pb.dim_ne_zero).mp (by simpa using pb.basis.mem_span y)
 
+lemma exists_eq_aeval' (pb : power_basis R S) (y : S) :
+∃ f : polynomial R, y = aeval pb.gen f :=
+begin
+  nontriviality S,
+  obtain ⟨f, _, hf⟩ := exists_eq_aeval pb y,
+  exact ⟨f, hf⟩
+end
+
+local attribute [instance] alg_hom.subsingleton
```

Comment:
It looks like you might not need this any more, since the `nontriviality` call is now higher up